pallets/motion/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+/// Dispatchable function benchmarks.
+mod benchmarking;
+
+/// Dispatchable function weights.
+pub mod weights;
+pub use pallet::*;
+use sp_runtime::DispatchResult;
+pub use weights::WeightInfo;
+
+extern crate alloc;
+use alloc::{boxed::Box, vec};
+
+/// The motion pallet.
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::GetDispatchInfo, pallet_prelude::*, traits::UnfilteredDispatchable,
+	};
+	use frame_system::pallet_prelude::*;
+
+	use super::{DispatchResult, *};
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The runtime event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The runtime call type.
+		type RuntimeCall: Parameter
+			+ UnfilteredDispatchable<RuntimeOrigin = Self::RuntimeOrigin>
+			+ GetDispatchInfo;
+		/// Origin that can act as `Root` origin if a collective has achieved a simple majority
+		/// consensus.
+		type SimpleMajorityOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Origin that can act as `Root` origin if a collective has achieved a super majority
+		/// consensus.
+		type SuperMajorityOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Origin that can act as `Root` origin if a collective has achieved a unanimous consensus.
+		type UnanimousOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Type representing the weight of this pallet
+		type WeightInfo: WeightInfo;
+	}
+
+	/// The events that can be emitted.
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A [SimpleMajorityOrigin] motion was executed. [motion_result] contains the call result
+		DispatchSimpleMajority { motion_result: DispatchResult },
+		/// A [SuperMajorityOrigin] motion was executed. [motion_result] contains the call result
+		DispatchSuperMajority { motion_result: DispatchResult },
+		/// A [UnanimousOrigin] motion was executed. [motion_result] contains the call result
+		DispatchUnanimous { motion_result: DispatchResult },
+	}
+
+	/// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	/// The dispatchable functions available.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Ensures the simple majority is met and dispatches a call with `Root` origin.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB write (event).
+		/// - Weight of derivative `call` execution + 10,000.
```

Comment:
What is the `+ 10,000`? Appreciate it wasnt written by you, but might as well question if still applicable and remove if not, otherwise docs are giving false information. Same applies for the others.

Reply:
Good question!

My understanding is that it represents the cost of wrapping the actual dispatch.  But would love if @peterwht could clarify this so I can incorporate that into the docs or remove it all together if it is not the case.

Reply:
It's been a couple years. But pallet-motion was similar to Sudo. If you go back to an older version of Sudo you can find this comment: https://github.com/paritytech/substrate/blob/monthly-2022-02/frame/sudo/src/lib.rs#L136

But now that we have proper benchmarking, the adding 10_000 shouldn't be necessary.